src/game/DBScripts/ScriptMgr.h

```diff
@@ -433,7 +433,7 @@ struct ScriptInfo
     float y;
     float z;
     float o;
-    uint32 condition_id;
+    uint32 condition_id = 0;
```

Comment:
It's a bit strange - and certainly not very consistent - to see this is the only member that's initialized here. I would assume the other members, all built-in types I think, also need some initialization. And I think that's done in ```ScriptMgr::LoadScripts()``` for the other members (whether that's a good idea or not is another discussion).

You have rightlfully adjusted ```ScriptMgr::LoadScripts()``` as well to initialize condition_id, so I'm wondering where this uninitialized condition_id was coming from? I'm going to guess it's related to ```ScriptInfo generateActivateCommand()``` in SpellEffects.cpp that you didn't update (but probably should).

Reply:
Yep, I made sure it is always initialized because of generateActivateCommand. When we first made this change on LH we had the displeasure of having to figure out why stuff is just randomly not working for seemingly no reason, and it turned out that a ScriptInfo can be generated from inside the core without initializing all members of the structure, thus leading to the condition_id sometimes being a random number preventing the script command from executing.

I thought it's a good idea to just make sure it is always initialized in all possible circumstances, saving future people the pain of debugging it if they decide to do what is done in generateActivateCommand and utilize a ScriptInfo object but forget to initialize the condition id.

Is it okay if I add a constructor to ScriptInfo and initialize all members for consistency? I think it's a better solution than just setting the condition to 0 inside generateActivateCommand, as it prevents this situation from occurring in the future as well. The C++ core guidelines says to always initialize your member variables.

Reply:
Yes, in my opinion it's certainly fine to add a constructor and initialize all members there. That's actually what I was referring to when I questioned whether initializing all members in ```ScriptMgr::LoadScripts()``` was a good idea. Because as you mention: any time you create a ScriptInfo object somewhere else, you'll basically start with random values for the members and forgetting to initialize one somewhere is too easy.

An alternative to writing a default constructor could be braced initialization of all the members, but we don't seem to do that often, so I'm fine with just a constructor as well.